src/Data/Version.php

```diff
@@ -0,0 +1,94 @@
+<?php declare(strict_types=1);
+
+namespace ILIAS\Data;
+
+/**
+ * A version number that consists of three numbers (major, minor, patch).
+ */
+class Version
+{
+    private const REGEXP = '(\d+)([.](\d+)([.](\d+))?)?';
+
+    protected int $major;
+    protected int $minor;
+    protected int $patch;
+
+    public function __construct(string $version)
+    {
+        $match = [];
+        if (!preg_match("/" . self::REGEXP . "/", $version, $match)) {
+            throw new \InvalidArgumentException(
+                "Expected version string to match this regular expression: " . self::REGEXP
+            );
+        }
+        $this->major = (int) $match[1];
+        $this->minor = 0;
+        $this->patch = 0;
+        if (count($match) > 3) {
```

Comment:
Just as an idea, you could use named groups in regex here. this would allow to use isset/empty. 
https://www.regular-expressions.info/refext.html

Reply:
Nice idea, thanks!